src/website/shared/listeners/automatic_linkage.py

```diff
@@ -10,19 +11,36 @@
 logger = logging.getLogger(__name__)
 
 
-def produce_linkage_candidates(container: Container) -> set[NixDerivation]:
+class ProvenanceFlags(IntFlag):
+    PACKAGE_NAME_MATCH = auto()
+    VERSION_CONSTRAINT_INRANGE = auto()
+    VERSION_CONSTRAINT_OUTOFRANGE = auto()
+    NO_SOURCE_VERSION_CONSTRAINT = auto()
+    # Whether the hardware constraint is matched for this derivation.
+    HARDWARE_CONSTRAINT_INRANGE = auto()
+    KERNEL_CONSTRAINT_INRANGE = auto()
```

Comment:
Why do we have that type twice, and the other is strict?

Reply:
Fixed.